lib/ex_subtil_backend/workflow/step/requirements.ex

```diff
@@ -0,0 +1,16 @@
+defmodule ExSubtilBackend.Workflow.Step.Requirements do
+
+  def get_required_paths(requirements \\ %{}, path)
+  def get_required_paths(requirements, paths) when is_list(paths) do
+    Map.update(requirements, :paths, paths, fn(cur_paths) ->
+      Enum.concat(cur_paths, paths)
+      |> Enum.uniq
+    end)
+  end
+  def get_required_paths(requirements, path) do
+    paths =
+        Map.get(requirements, :paths, [])
+        |> List.insert_at(-1, path)
+    get_required_paths(requirements, paths)
```

Comment:
Why ?
You're shure to not loop indefinitely ?

Reply:
No, because here `paths` is always a list, so it calls the other [get_required_paths](https://github.com/FTV-Subtil/ex_subtil_backend/pull/50/files/d34b2d372ed54304a0257b482ed3b66f2ccc5b29#diff-12b4f7e61a84190d70a75c47299c8a2cR4) function